### DISCUSSION TITLE
anytype-ts: reasons for retaining use of class components?

### DISCUSSION THREAD

**Discussion by pstashp** [anyproto/contributors#167] [Q&A]
Greetings :)

I understand that there are pros and cons to every decision, but I'm curious on the team's reasoning in continuing the use of React class components.

My team is in a dilemma on whether to make a complete switch.  With AnyType being one of the few code-bases that are still on class components, it would be nice if you guys can share your wisdom.

Thanks.

**Comment by ra3orblade** [comment]
There is no wisdom behind it, we use class components because the codebase is huge and switching completely will take a lot of time and I personally do not like hooks, and find the very hard to read them. But I have a plan switching to react@18 and hooks after it.
